api/migrations/20170717222139_add_email_template_table.js

```diff
@@ -0,0 +1,28 @@
+exports.up = function(knex, Promise) {
+  const addTemplateTable = knex.schema.createTable('SIP_client_email_templates', table => {
+  	table.uuid('id').primary();
+  	table.text('template');
+  });
+
+  const addClientTemplateReference = knex.schema.table('SIP_client', table => {
+  	table.uuid('reset_password_template_id').references('id').inTable('SIP_client_email_templates').onDelete('CASCADE').onUpdate('CASCADE');
+  });
+
+  return Promise.all([
+  	addTemplateTable,
+  	addClientTemplateReference

```

Comment:
`Promise.all()` can't guarantee that your promises execute sequentially. 

Since your second query requires your first query to be done first your second query should be done inside the `.then()` function of the first.

Also, can you rename the table to `SIP_templates`? This table will be used for non-email templates and since there's no client id stored here anymore the client prefix is sort of limiting.